jgrapht-core/src/main/java/org/jgrapht/alg/matching/GreedyWeightedMatching.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * (C) Copyright 2016-2016, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.matching;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.jgrapht.Graph;
+import org.jgrapht.alg.interfaces.WeightedMatchingAlgorithm;
+import org.jgrapht.alg.util.Pair;
+
+/**
+ * The greedy algorithm for computing a maximum weight matching in an arbitrary graph. The algorithm
+ * is a 1/2-approximation algorithm and runs in O(n + m log n) where n is the number of vertices and
+ * m is the number of edges of the graph. This implementation accepts directed and undirected graphs
+ * which may contain self-loops and multiple edges.
+ * 
+ * <p>
+ * The greedy algorithm is the algorithm that first orders the edge set in non-increasing order of
+ * weights and then greedily constructs a maximal matching using this ordering. A maximal matching
+ * (not to be confused with maximum) is a matching that cannot be increased in cardinality without
+ * removing an edge first.
+ *
+ * <p>
+ * For more information about approximation algorithm for the maximum weight matching problem in
+ * arbitrary graphs see:
+ * <ul>
+ * <li>R. Preis, Linear Time 1/2-Approximation Algorithm for Maximum Weighted Matching in General
+ * Graphs. Symposium on Theoretical Aspects of Computer Science, 259-269, 1999.</li>
+ * <li>D.E. Drake, S. Hougardy, A Simple Approximation Algorithm for the Weighted Matching Problem,
+ * Information Processing Letters 85, 211-213, 2003.</li>
+ * </ul>
+ * 
+ * @see PathGrowingWeightedMatching
+ * 
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ * 
+ * @author Dimitrios Michail
+ * @since September 2016
+ */
+public class GreedyWeightedMatching<V, E>
+    implements WeightedMatchingAlgorithm<V, E>
```

Comment:
I agree with @jsichi 's remark. This is also valid for the flow algorithms for example.